data/data-pipeline/data_pipeline/etl/sources/cdc_life_expectancy/etl.py

```diff
@@ -23,32 +35,122 @@ def __init__(self):
+            set(self.STATE_FIPS_CODES)
+            # We don't expect LEEP to have data for island areas or Puerto Rico.
+            - set(TILES_ISLAND_AREA_FIPS_CODES)
+            - set(TILES_PUERTO_RICO_FIPS_CODE)
+        )
+
+        # These states are currently missing from LEEP's whole USA file.
+        self.EXPECTED_MISSING_STATES = ["23", "55"]
+
         self.raw_df: pd.DataFrame
         self.output_df: pd.DataFrame
 
     def extract(self) -> None:
         logger.info("Starting data download.")
 
-        download_file_name = (
+        all_usa_download_file_name = (
             self.get_tmp_path() / "cdc_life_expectancy" / "usa.csv"
         )
         download_file_from_url(
-            file_url=self.FILE_URL,
-            download_file_name=download_file_name,
+            file_url=self.USA_FILE_URL,
+            download_file_name=all_usa_download_file_name,
+            verify=True,
+        )
+
+        pandas_read_csv_dtype_settings = {
+            # The following need to remain as strings for all of their digits, not get converted to numbers.
+            self.TRACT_INPUT_COLUMN_NAME: "string",
+            self.STATE_INPUT_COLUMN_NAME: "string",
+        }
+
+        all_usa_raw_df = pd.read_csv(
+            filepath_or_buffer=all_usa_download_file_name,
+            dtype=pandas_read_csv_dtype_settings,
+            low_memory=False,
+        )
+
+        # Check which states are missing
+        states_in_life_expectancy_usa_file = all_usa_raw_df[
+            self.STATE_INPUT_COLUMN_NAME
+        ].unique()
+
+        # Find which states are missing from the expected set.
+        states_missing = sorted(
+            list(
+                self.EXPECTED_STATES_SET
+                - set(states_in_life_expectancy_usa_file)
+            )
+        )
+
+        if states_missing != self.EXPECTED_MISSING_STATES:
+            raise ValueError(
+                "LEEP data has changed. The states missing from the data are "
+                "no longer the same."
+            )
+
+        logger.info("Downloading data for Maine")
+        maine_download_file_name = (
+            self.get_tmp_path() / "cdc_life_expectancy" / "maine.csv"
+        )
+        download_file_from_url(
+            file_url=self.MAINE_FILE_URL,
+            download_file_name=maine_download_file_name,
             verify=True,
         )
+        maine_raw_df = pd.read_csv(
+            filepath_or_buffer=maine_download_file_name,
+            dtype=pandas_read_csv_dtype_settings,
+            low_memory=False,
+        )
 
-        self.raw_df = pd.read_csv(
-            filepath_or_buffer=download_file_name,
-            dtype={
-                # The following need to remain as strings for all of their digits, not get converted to numbers.
-                self.TRACT_INPUT_COLUMN_NAME: "string",
-            },
+        logger.info("Downloading data for Wisconsin")
+        wisconsin_download_file_name = (
```

Comment:
nit: can we make these three steps into a single function call? ("download and prep data" or something?) It might be DRYer?